src/main/java/seedu/address/logic/commands/commandAssign/AssignStudentToCourseCommand.java

```diff
@@ -42,51 +43,39 @@ public static boolean isValidDescriptor(AssignDescriptor assignDescriptor) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException, ParseException {
-        String courseidString = this.assignDescriptor.getAssignID(PREFIX_COURSEID).value;
-        String studentidString = this.assignDescriptor.getAssignID(PREFIX_STUDENTID).value;
-        String courseName = "";
-        String studentName = "";
+        // if student exists
+        // if course exists
+        // if student not already assigned to the course
+        // if course doesn't already have the student
 
-        boolean courseExists = false;
-        boolean studentExists = false;
-        Course foundCourse = null;
-        Student foundStudent = null;
-
-        for (Course course : model.getFilteredCourseList()) {
-            if (course.getId().value.equals(courseidString)) {
-                courseName = course.getName().toString();
-                courseExists = true;
-                foundCourse = course;
-                break;
-            }
-        }
+        ID courseID = this.assignDescriptor.getAssignID(PREFIX_COURSEID);
+        ID studentID = this.assignDescriptor.getAssignID(PREFIX_STUDENTID);
+        HashMap<String, ID> progressIDConstructor = new HashMap<>();
 
-        for (Student student : model.getFilteredStudentList()) {
-            if (student.getID().value.equals(studentidString)) {
-                studentName = student.getName().toString();
-                studentExists = true;
-                foundStudent = student;
-                break;
-            }
-        }
+        boolean courseExists = model.hasCourse(courseID);
+        boolean studentExists = model.hasStudent(studentID);
+
+        String courseName = "";
+        String studentName = "";
 
         if (!courseExists) {
             throw new CommandException(MESSAGE_INVALID_COURSE_ID);
         } else if (!studentExists) {
             throw new CommandException(MESSAGE_INVALID_STUDENT_ID);
         } else {
-            ID courseid = ParserUtil.parseCourseid(courseidString);
-            ID studentid = ParserUtil.parseStudentid(studentidString);
-            foundCourse.addStudent(studentid);
-            foundStudent.addCourse(courseid);
+            Course foundCourse = model.getCourse(courseID);
+            Student foundStudent = model.getStudent(studentID);
+
+            foundCourse.addStudent(studentID);
+            foundStudent.addCourse(courseID);
```

Comment:
why do we still need this? I thought we center the data to Course class

Reply:
So it means no need to add to student class

Reply:
Okay, the thing is I havent' finished implementing this part. Second thing is I thought really long about it and I think we should have a list of courseID in student. Please see here for diagram

![photo6060082817980803656](https://user-images.githubusercontent.com/58815457/77985756-e03a2280-7347-11ea-9895-64c3b1736992.jpg)